utils.py

```diff
@@ -48,6 +53,20 @@
 SECONDS_IN_HOUR = 60 * 60
 SECONDS_IN_MINUTE = 60
 
+T = TypeVar('T')
+U = TypeVar('U')
```

Comment:
Why are these needed?

Reply:
Lets see this example. 
```
def get_random_choice(alist):
    # type: (List[T]) -> T
```
So what this function does is that it returns a random element from the list. This means that a list of integers must return an integer and so on.
Now lets say we have written `# type: (List[Any]) -> Any`, this means a list is given as a parameter and anything can be returned.
So I think the earlier type statement is more strict.

Reply:
Agreed, just that it seems a bit weird to me that we have to define these, cannot we use `# type: (List[T]) -> T` without these two lines?

Reply:
No, we cannot use this directly. Mypy is unable to determine what is `T` here. Yes this seems weird but this is used by other organisations as well. [Here](https://github.com/zulip/zulip/blob/12a439b8bd4f5c8f6c54dcb926e48d4d71d81792/zerver/lib/pysa.py) is an example.

Reply:
Okay, I went through all the types in this file and we do not use `U` so you can probably remove it.

Reply:
Removed.